ShopperInsights/src/main/java/com/braintreepayments/api/shopperinsights/ShopperInsightsClient.kt

```diff
@@ -159,61 +165,60 @@ class ShopperInsightsClient internal constructor(
+     * @param presentmentDetails Detailed information, including button order, experiment type,
+     * and page type about the payment button that is sent to analytics to help improve the Shopper
+     * Insights feature experience.
      */
-    fun sendPayPalPresentedEvent(
-        experiment: String? = null,
-        paymentMethodsDisplayed: List<String> = emptyList()
+    fun sendPresentedEvent(
+        buttonType: ButtonType,
+        presentmentDetails: PresentmentDetails
     ) {
-        braintreeClient.sendAnalyticsEvent(
-            PAYPAL_PRESENTED,
-            AnalyticsEventParams(
-                experiment = experiment,
-                paymentMethodsDisplayed = paymentMethodsDisplayed
-            )
+        val params = AnalyticsEventParams(
+            experiment = presentmentDetails.type?.formattedExperiment(),
+            shopperSessionId = shopperSessionId,
+            buttonType = buttonType.getStringRepresentation(),
+            buttonOrder = presentmentDetails.buttonOrder.getStringRepresentation(),
+            pageType = presentmentDetails.pageType.getStringRepresentation()
         )
+
+        braintreeClient.sendAnalyticsEvent(BUTTON_PRESENTED, params)
     }
 
     /**
-     * Call this method when the PayPal button has been selected/tapped by the buyer.
+     * Call this method when the PayPal, Venmo or Other button has been selected/tapped by the buyer.
      * This method sends analytics to help improve the Shopper Insights feature experience.
+     * @param buttonType Type of button presented - PayPal, Venmo, or Other.
      */
-    fun sendPayPalSelectedEvent() {
-        braintreeClient.sendAnalyticsEvent(PAYPAL_SELECTED)
+    fun sendSelectedEvent(
+        buttonType: ButtonType,
+    ) {
+        val params = AnalyticsEventParams(
+            shopperSessionId = shopperSessionId,
+            buttonType = buttonType.getStringRepresentation(),
+        )
+
+        braintreeClient.sendAnalyticsEvent(BUTTON_SELECTED, params)
     }
 
     /**
-     * Call this method when the Venmo button has been successfully displayed to the buyer.
-     * This method sends analytics to help improve the Shopper Insights feature experience.
-     *
-     * @param experiment optional JSON string representing an experiment you want to run
-     * @param paymentMethodsDisplayed optional The list of available payment methods,
-     * rendered in the same order in which they are displayed
+     * Indicates whether the PayPal App is installed.
      */
-    fun sendVenmoPresentedEvent(
-        experiment: String? = null,
-        paymentMethodsDisplayed: List<String> = emptyList()
-    ) {
-        braintreeClient.sendAnalyticsEvent(
-            VENMO_PRESENTED,
-            AnalyticsEventParams(
-                experiment = experiment,
-                paymentMethodsDisplayed = paymentMethodsDisplayed
-            )
-        )
+    fun isPayPalAppInstalled(context: Context): Boolean {
```

Comment:
Quick ? - should these methods be marked beta too?

Reply:
I think that should be unnecessary since the whole class is market as experimental